### ISSUE TITLE
Error when establishing JSTP connection before full start

### ISSUE THREAD

**Issue by belochub** [metarhia/Impress#726]
Stack trace:
```
2017-04-21T14:11:05.800Z        Worker(18883/S1N3)      [impress]
TypeError: Cannot read property 'length' of undefined;
Object.api.common.generateSID (api.common.js:646:20);
Server.impress.jstp.startSession (impress.jstp.js:117:30);
Connection._processHandshakeRequest (/metarhia-jstp/lib/connection.js:402:15);
Connection._processHandshakePacket (/metarhia-jstp/lib/connection.js:365:10);
Connection._processPacket (/metarhia-jstp/lib/connection.js:336:13);
emitOne (events.js:96:13);
TcpTransport.emit (events.js:188:7);
TcpTransport._onSocketData (/metarhia-jstp/lib/transport.tcp.js:239:10);
emitOne (events.js:96:13);
TLSSocket.emit (events.js:188:7);
readableAddChunk (_stream_readable.js:176:18);
TLSSocket.Readable.push (_stream_readable.js:134:10);
TLSWrap.onread (net.js:548:20)
```
`impress version` output:
```
Impress Application Server 0.3.36
 Impress AS: 0.3.36
    Node.js: 6.10.0
         v8: 5.1.281.93
      libuv: 1.9.1
       zlib: 1.2.8
   Open SSL: 1.0.2k
HTTP parser: 2.7.0
         OS: Linux 3.10.0-327.18.2.el7.x86_64 x64
```
`metarhia-jstp@0.5.0`
@tshemsedinov, I am not sure if this was fixed in Impress version 0.4.x, take a look, please.
/cc @aqrln

**Comment by tshemsedinov** [comment]
#727 and #728

**Comment by tshemsedinov** [comment]
Fixed for both active branches `v0.3` and `v0.4`
